components/DogCard.js

```diff
@@ -1,29 +1,115 @@
+import React, { useState, useEffect } from "react";
 import styled from "styled-components";
-import { useState } from "react";
+import Image from "next/image";
+import editButtonCardImage from "../public/images/edit-button-card.png";
+
+export default function DogCard({ dog, onDelete, onEdit, onUpdate }) {
```

Comment:
```suggestion
export default function DogCard({ dog, onDelete, onUpdate }) {
```
You should remove the `onEdit` prop, since it is not used in this file.